### PR TITLE
[IN-103] Make Home page that displays all job postings from other recruiters

### PR DESCRIPTION
**Pull request by tommytwm**
#### User Story
[[IN-103] Make Home page that displays all job postings from other recruiters](https://chaosneutral.atlassian.net/browse/IN-103)

#### Changes made
- Made a new default landing home page for authenticated users
- Removed `Browse` link on authenticated nav bar
- Added new `TableUser` that takes in all job posting data and allows for pagination
- Added two new fields `authorName` and `profilePicture` to job posting schema
  - *NOTES:*
    - I initially had the designs fill in the handle name for the users but I accidentally put author name and I guess that works too. Eventually, we would want to have functionality where we would be able to access other people's profiles via the table or view page as well.
    - Right now the job posting collection does not have any author names in it so we will need to clean out ALL of our job postings and repopulate them with proper accounts

https://user-images.githubusercontent.com/37882255/128553614-4529d8f3-26ac-4042-8823-b68f5e9838d9.mov

#### Does your new code introduce new warnings on dev console?
Nothing in my console, but best to double check.

#### Test Steps
1. Log on to Internify
2. You should be prompted login to the home page

### CODE REVIEW

File: client/src/store/store.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ const middleware = [thunk];
 
 const store = createStore(rootReducer, initialState, compose(
     applyMiddleware(...middleware),
-    // window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
+    window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
```
Comment (JudeShamsi) [inline client/src/store/store.js]: comment out

File: client/src/components/molecules/Table.js [hunk 2]
```diff
@@ -24,10 +37,21 @@ function createBasicData(jobId, title, dateCreated, matches) {
   return { jobId, title, dateCreated, matches };
 }
 
-function createData(isStarred, jobId, title, dateCreated, matches) {
+function createStarredData(isStarred, jobId, title, dateCreated, matches) {
   return { isStarred, jobId, title, dateCreated, matches };
```
Comment (JudeShamsi) [inline client/src/components/molecules/Table.js]: do we no longer want to return isStarred?

File: client/src/components/pages/Home.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+          <Grid item className="home_header_bio">
+            <h1 style={{ marginTop: 0 }}>Hi, {firstName}</h1>
+            <p style={{ marginBottom: 0 }}>
+              Take a look at the community's postings!
+            </p>
+          </Grid>
+        </Grid>
+
+        <Grid item style={{ paddingLeft: "15%"}}>
+            <TableUsers data={props.jobs.currentListOfJobs} />
```
Comment (JudeShamsi) [inline client/src/components/pages/Home.js]: is currentListOfJobs supposed to hold jobs related to a specific user or related to all job postings in the database? bc I recall it's being used on profile page to list out the jobs associated with a user. Maybe to be safe, change this to use a new list -> 'allJobPostingsList' instead.

File: client/src/store/actions/jobPostActions.js [hunk 4]
```diff
@@ -30,6 +30,19 @@ export const getJobs = (user) => (dispatch) => {
         .catch((err) => console.error(err));
 };
 
+export const getAllJobs = () => (dispatch) => {
+      axios
+        .get("/api/jobs")
+        .then((res) => {
+          console.log(res.data);
```
Comment (JudeShamsi) [inline client/src/store/actions/jobPostActions.js]: remove console.log

**Review by JudeShamsi — CHANGES_REQUESTED** [review]
see comments

File: client/src/components/molecules/Table.js (hunk 2 above)
Comment (tommytwm) [reply client/src/components/molecules/Table.js]: This table was initially designed for the intention of having bookmarked (starred) job postings. Considering we aren't using this. I'm just gonna remove it

**Comment by ReneReid** [comment]
wow - this is really wonderful, @tommytwm !

**Review by ReneReid — APPROVED** [review]

**Review by ghost — CHANGES_REQUESTED** [review]
The three bubbles end up following the scrollbar down, I'm wondering if there's any way to keep it fixed at the top. It interferes with seeing the other postings by users as you scroll. 

Also, I checked in the new JobPostingData model, and noticed there is an author field as well as authorName, is there a reason for the distinction?

**Review by JudeShamsi — APPROVED** [review]

**Pull request merged**